[CI] Sandbox error when running shards testuite on GitHub Actions

> I finally managed to get some more useful error messages, for the failure at https://github.com/JuliaPackaging/BinaryBuilderBase.jl/blob/8d1eca1e3a63a2de30d9a9eee7381c9ca999ea60/test/runners.jl#L446
> ```
> rustc -o /tmp/testsuite/x86_64-linux-gnu/rust/hello_world/hello_world hello_world.rs
>   Installing cargo_build v0.1.0 (/usr/share/testsuite/rust/cargo_build)
> error: failed to compile `cargo_build v0.1.0 (/usr/share/testsuite/rust/cargo_build)`, intermediate artifacts can be found at `/usr/share/testsuite/rust/cargo_build/target`.
> To reuse those artifacts with a future compilation, set the environment variable `CARGO_TARGET_DIR` to that path.
> 
> Caused by:
>   Cross-device link (os error 18)
> make[1]: *** [Makefile:7: /tmp/testsuite/x86_64-linux-gnu/rust/cargo_build/cargo_build] Error 101
> make: *** [Makefile:48: install-project-rust-cargo_build] Error 2
> ```
> Sounds like a problem in the sandbox code, which perhaps only arises on the CI setup.

_Originally posted by @giordano in https://github.com/JuliaPackaging/BinaryBuilderBase.jl/issues/346#issuecomment-1793431598_

Note: this happens only with the non-squashfs images with UserNS runners (not Docker).

--- a/test/runners.jl
+++ b/test/runners.jl
@@ -545,10 +545,7 @@ end
             set -e
             make -j${nproc} -sC /usr/share/testsuite install
             """
-            # This test fails on GitHub Actions with non-squashfs on UserNS runners:
-            # <https://github.com/JuliaPackaging/BinaryBuilderBase.jl/issues/347>.
-            is_broken = get(ENV, "GITHUB_ACTIONS", "false")=="true" && !(BinaryBuilderBase.use_squashfs[]) && BinaryBuilderBase.preferred_runner() == BinaryBuilderBase.UserNSRunner
-            @test run(ur, `/bin/bash -c "$(test_script)"`, iobuff) broken=is_broken
+            @test run(ur, `/bin/bash -c "$(test_script)"`, iobuff)
         end
     end
 end

